clientconn.go

```diff
@@ -1032,6 +1032,7 @@ func (ac *addrConn) resetTransport() {
 				continue
 			}
 
+			backoffFor = 0
```

Comment:
It works fine for us to reconnect immediately after connection closed.
But I think this also resets backoff duration when reconnecting failed repeated in the server down, which resulted in trying to reconnect aggressively. 

So the backoff duration should be reset only in first retry like this?
```
if ac.backoffIdx == 0 {
	backoffFor = 0
}
```

Reply:
If reconnecting failed, we will `continue` above, and won't reach this.
This only happens when we successfully create a connection, and we want to skip any backoff if this connection breaks.

Reply:
sorry it's my bad. I tried again with this branch and it does retry to connect with backoff correctly.